presto-feature-toggle/README.md

```diff
@@ -0,0 +1,233 @@
+                .bind();
+        }
+    }
+```
+
+Toggle strategy configuration params can be updated on runtime, by changing strategy configuration param values.
+
+In this case, we can change `feature.query-cancel.strategy.allow-list-source` and `feature.query-cancel.strategy.allow-list-user` param values.
+
+# Examples
+
+## Query Cancel Feature
+
+```
+    feature.query-cancel.enable=true
+    feature.query-cancel.strategy=AllowList
+    feature.query-cancel.strategy.allow-list-source=.*IDEA.*
+    feature.query-cancel.strategy.allow-list-user=.*prestodb
+```
+
+Feature Toggle Strategies are evaluated each time we check if the feature toggle is enabled.
+The result of the Feature Toggle Strategy evaluation overrides the `enabled` status of the Feature Toggle.
+
+Simple Feature toggle check Example:
+
+```
+    class ClassWithQueryCancel
+    {
+        private final Supplier<Boolean> isQueryCancelEnabled;
+        private final Function<Object, Boolean> isQueryCancelEnabledForQueryId;
+
+        @Inject
+        public SupplierInjectionRunner(
+            @FeatureToggle("query-cancel") Supplier<Boolean> isQueryCancelEnabled,
+            @FeatureToggle("FunctionInjectionFeature") Function<Object, Boolean> isQueryCancelEnabledForQueryId)
+        {
+            this.isQueryCancelEnabledForQueryId = isQueryCancelEnabledForQueryId;
+            this.isQueryCancelEnabled = isQueryCancelEnabled;
+        }
+
+        /**
+        * simple check: checks enabled param of the configuration
+        */
+        public boolean testSimpleFeatureEnabled()
+        {
+            return isQueryCancelEnabled.get();
+        }
+
+        /**
+        * feature toggle check using a strategy that accepts input param
+        */
+         public boolean testFunctionInjectionFeatureEnabled(String queryId)
+        {
+            return isQueryCancelEnabledForQueryId.apply(queryId);
+        }
+    }
+```
+
+## Feature Toggle config file example
+
+Any change of the configuration in the configuration file (source) overrides the feature toggle configuration.
+Allowed configuration params are:
+
+- enable
+- strategy.<param>
+
+```
+# feature.query-cancel
+feature.query-cancel.enable=true
+feature.query-cancel.strategy=AllowList
+feature.query-cancel.strategy.allow-list-source=.*IDEA.*
+feature.query-cancel.strategy.allow-list-user=.*prestodb
+```
+
+Configuration properties always start with a `feature` followed by a dot and the feature id.
+
+Feature Toggle strategy properties start with `feature.featureId.strategy`. Property `feature.featureId.strategy` defines feature toggle strategy class (declared by registered
+name).
+After that, we can declare key-value pairs for parameters allowed for a given strategy.
+
+In this example for feature `query-cancel`, changing the value of feature.query-cancel.enabled to `false` will 'disable' this feature.
+Changes will be effective within the refresh period. 
```

Comment:
```suggestion
Changes are effective within the refresh period. 
```
Does "within" mean the change is effective during the current refresh period, or after the refresh period?